### PR TITLE
feat: add readOnlyRootFilesystem if possible

### PR DESCRIPTION
**Pull request by farodin91**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
add securityContext for pod and container
```

### CODE REVIEW

File: charts/latest/csi-driver-nfs/templates/csi-nfs-node.yaml [hunk 1]
```diff
@@ -79,12 +83,11 @@ spec:
             - name: registration-dir
               mountPath: /registration
           resources: {{- toYaml .Values.node.resources.nodeDriverRegistrar | nindent 12 }}
-        - name: nfs
+          {{- with .Values.node.containerSecurityContext.nodeDriverRegistrar }}
           securityContext:
```
Comment (andyzhangx) [inline charts/latest/csi-driver-nfs/templates/csi-nfs-node.yaml:87]: currently only `nfs` container needs `securityContext`, I don't think it's necessary to set `securityContext` for every sidecar container

Comment (farodin91) [reply charts/latest/csi-driver-nfs/templates/csi-nfs-node.yaml:87]: We want to set readonlyrootfilesystem. You have set it on every container.

Comment (andyzhangx) [reply charts/latest/csi-driver-nfs/templates/csi-nfs-node.yaml:87]: then just set `readOnlyRootFilesystem` for other containers by default, and only need to set `.Values.node.securityContext` and `.Values.controller.securityContext` for nfs container.

Comment (farodin91) [reply charts/latest/csi-driver-nfs/templates/csi-nfs-node.yaml]: I changed the behavior

**Comment by andyzhangx** [comment]
seems it does not work:
```
failed to provision volume with StorageClass "nfs-7201-nfs.csi.k8s.io-dynamic-sc-p4hsv": rpc error: code = Internal desc = failed to mount nfs server: rpc error: code = Internal desc = mkdir /tmp/pvc-0d199297-cb02-4cf3-b2d4-f1deeb878d82: read-only file system
```

**Comment by farodin91** [comment]
@andyzhangx i did some refactoring

**Comment by farodin91** [comment]
@andyzhangx  chart.tar.gz updated

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Pull request merged**